lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb

```diff
@@ -118,8 +118,8 @@ def self.create_service_provision_request(svc_template, options = nil)
     end
 
     def self.create_retire_request(obj)
-      result = obj.object_send(:make_retire_request, User.current_user)
-      MiqAeServiceModelBase.wrap_results(result)
+      obj_class = "#{obj.object_class.base_model.name}".constantize
```

Comment:
While Robocop has a good point with preferring`to_s over string interpolation` the reality is `.name` will return a string so there is no need for either.

`obj_class = obj.object_class.base_model.name.constantize`